### ISSUE TITLE
Cannot assign Pimcore\Model\Element\ElementDescriptor to property Pimcore\Model\DataObject\Objectbrick::$object

### ISSUE THREAD

**Issue by piotrecky** [pimcore/pimcore#10239]
### Expected behavior

Pimcore cannot write and read the object with object bricks

### Actual behavior

PHP Fatal error:  Uncaught TypeError: Cannot assign Pimcore\Model\Element\ElementDescriptor to property Pimcore\Model\DataObject\Objectbrick::$object of type ?Pimcore\Model\DataObject\Concrete in /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php:253
Stack trace:
#0 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(253): ReflectionProperty->setValue(Object(Pimcore\Model\DataObject\Produkt\Parametry), Object(Pimcore\Model\Element\ElementDescriptor))
#1 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy->copyObjectProperty(Object(Pimcore\Model\DataObject\Produkt\Parametry), Object(ReflectionProperty))
#2 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy->copyObject(Object(Pimcore\Model\DataObject\Produkt\Parametry))
#3 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(253): DeepCopy\DeepCopy->recursiveCopy(Object(Pimcore\Model\DataObject\Produkt\Parametry))
#4 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy->copyObjectProperty(Object(Pimcore\Model\DataObject\Data\BlockElement), Object(ReflectionProperty))
#5 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy->copyObject(Object(Pimcore\Model\DataObject\Data\BlockElement))
#6 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(94): DeepCopy\DeepCopy->recursiveCopy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#7 /var/www/vendor/pimcore/pimcore/models/DataObject/Data/BlockElement.php(246): DeepCopy\DeepCopy->copy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#8 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(564): Pimcore\Model\DataObject\Data\BlockElement->marshalForCache()
#9 [internal function]: Pimcore\Cache\Core\CoreCacheHandler->Pimcore\Cache\Core\{closure}(Object(Pimcore\Model\DataObject\Data\BlockElement))
#10 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/TypeFilter/ReplaceFilter.php(28): call_user_func(Object(Closure), Object(Pimcore\Model\DataObject\Data\BlockElement))
#11 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(125): DeepCopy\TypeFilter\ReplaceFilter->apply(Object(Pimcore\Model\DataObject\Data\BlockElement))
#12 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(155): DeepCopy\DeepCopy->recursiveCopy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#13 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(135): DeepCopy\DeepCopy->copyArray(Array)
#14 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(155): DeepCopy\DeepCopy->recursiveCopy(Array)
#15 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(135): DeepCopy\DeepCopy->copyArray(Array)
#16 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(253): DeepCopy\DeepCopy->recursiveCopy(Array)
#17 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy->copyObjectProperty(Object(Pimcore\Model\DataObject\Objectbrick\Data\KredytGotowkowy), Object(ReflectionProperty))
#18 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy->copyObject(Object(Pimcore\Model\DataObject\Objectbrick\Data\KredytGotowkowy))
#19 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(253): DeepCopy\DeepCopy->recursiveCopy(Object(Pimcore\Model\DataObject\Objectbrick\Data\KredytGotowkowy))
#20 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy->copyObjectProperty(Object(Pimcore\Model\DataObject\Produkt\Parametry), Object(ReflectionProperty))
#21 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy->copyObject(Object(Pimcore\Model\DataObject\Produkt\Parametry))
#22 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(253): DeepCopy\DeepCopy->recursiveCopy(Object(Pimcore\Model\DataObject\Produkt\Parametry))
#23 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy->copyObjectProperty(Object(App\Model\DataObject\Product), Object(ReflectionProperty))
#24 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy->copyObject(Object(App\Model\DataObject\Product))
#25 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(94): DeepCopy\DeepCopy->recursiveCopy(Object(App\Model\DataObject\Product))
#26 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(575): DeepCopy\DeepCopy->copy(Object(App\Model\DataObject\Product))
#27 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(880): Pimcore\Cache\Core\CoreCacheHandler->storeCacheData('object_1163', Object(App\Model\DataObject\Product), Array, NULL, false)
#28 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(928): Pimcore\Cache\Core\CoreCacheHandler->writeSaveQueue()
#29 /var/www/vendor/pimcore/pimcore/lib/Cache.php(200): Pimcore\Cache\Core\CoreCacheHandler->shutdown(false)
#30 /var/www/vendor/pimcore/pimcore/lib/Pimcore.php(242): Pimcore\Cache::shutdown()
#31 /var/www/vendor/pimcore/pimcore/lib/Kernel.php(247): Pimcore::shutdown()
#32 [internal function]: Pimcore\Kernel->Pimcore\{closure}()
#33 {main}
  thrown in /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php on line 253

### Steps to reproduce

save object contains object brick with some elements

**Comment by dvesh3** [comment]
@piotrecky sorry, not able to reproduce the issue. please provide more specific steps to reproduce (like what elements to use inside object brick) and Pimcore version. thanks!

**Comment by piotrecky** [comment]
on pimcore demo no errors are visible  - the prod environment variable is probably set. Pimcore 10.1.2, redis on.

**Comment by piotrecky** [comment]
I have the analysis progress. When I disable cache in CoreCacheHandler everything works fine. The problem is likely to be in the marshalForCache method

**Comment by piotrecky** [comment]
reproduce steps:

1. Install demo from https://github.com/pimcore/demo
2. Create class "testobject" with objectbricks data component
3. Create Objectbrick "testbrick" with block data component "parameters" and one input inside block
4. Set testbrick allowed class to testobject
5. Create object with complete data and save.
6. Reopen object and see to console:

<b>Fatal error</b>:  Uncaught TypeError: Cannot assign Pimcore\Model\Element\ElementDescriptor to property Pimcore\Model\DataObject\Objectbrick::$object of type ?Pimcore\Model\DataObject\Concrete in /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php:252
Stack trace:
#0 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): ReflectionProperty-&gt;setValue(Object(Pimcore\Model\DataObject\Testobject\Parameters), Object(Pimcore\Model\Element\ElementDescriptor))
#1 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Testobject\Parameters), Object(ReflectionProperty))
#2 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Testobject\Parameters))
#3 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Testobject\Parameters))
#4 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Data\BlockElement), Object(ReflectionProperty))
#5 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Data\BlockElement))
#6 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(94): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#7 /var/www/vendor/pimcore/pimcore/models/DataObject/Data/BlockElement.php(246): DeepCopy\DeepCopy-&gt;copy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#8 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(564): Pimcore\Model\DataObject\Data\BlockElement-&gt;marshalForCache()
#9 [internal function]: Pimcore\Cache\Core\CoreCacheHandler-&gt;Pimcore\Cache\Core\{closure}(Object(Pimcore\Model\DataObject\Data\BlockElement))
#10 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/TypeFilter/ReplaceFilter.php(28): call_user_func(Object(Closure), Object(Pimcore\Model\DataObject\Data\BlockElement))
#11 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(125): DeepCopy\TypeFilter\ReplaceFilter-&gt;apply(Object(Pimcore\Model\DataObject\Data\BlockElement))
#12 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(155): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#13 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(135): DeepCopy\DeepCopy-&gt;copyArray(Array)
#14 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(155): DeepCopy\DeepCopy-&gt;recursiveCopy(Array)
#15 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(135): DeepCopy\DeepCopy-&gt;copyArray(Array)
#16 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy-&gt;recursiveCopy(Array)
#17 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Objectbrick\Data\Testbrick), Object(ReflectionProperty))
#18 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Objectbrick\Data\Testbrick))
#19 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Objectbrick\Data\Testbrick))
#20 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Testobject\Parameters), Object(ReflectionProperty))
#21 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Testobject\Parameters))
#22 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Testobject\Parameters))
#23 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Testobject), Object(ReflectionProperty))
#24 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Testobject))
#25 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(94): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Testobject))
#26 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(575): DeepCopy\DeepCopy-&gt;copy(Object(Pimcore\Model\DataObject\Testobject))
#27 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(880): Pimcore\Cache\Core\CoreCacheHandler-&gt;storeCacheData('object_3', Object(Pimcore\Model\DataObject\Testobject), Array, NULL, false)
#28 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(928): Pimcore\Cache\Core\CoreCacheHandler-&gt;writeSaveQueue()
#29 /var/www/vendor/pimcore/pimcore/lib/Cache.php(200): Pimcore\Cache\Core\CoreCacheHandler-&gt;shutdown(false)
#30 /var/www/vendor/pimcore/pimcore/lib/Pimcore.php(242): Pimcore\Cache::shutdown()
#31 /var/www/vendor/pimcore/pimcore/lib/Kernel.php(247): Pimcore::shutdown()
#32 [internal function]: Pimcore\Kernel-&gt;Pimcore\{closure}()
#33 {main}
  thrown in <b>/var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php</b> on line <b>252</b><br />

**Comment by aclDeveloper** [comment]
We are experiencing similar behavior when adding Blocks to Object Bricks. There also occurs an exception in the DeepCopy library.

```
Error: You're trying to decode an invalid JSON String: {"idPath":"\/1\/13\/14","hasPreview":false,"general":{"o_classId":"brand","o_className":"Brand","o_published":true,"o_id":14,"o_parentId":13,"o_type":"object","o_key":"Xbox","o_creationDate":1630566211,"o_userOwner":2,"fullpath":"\/Brands\/Xbox","o_locked":false,"php":{"classes":["Pimcore\\Model\\DataObject\\Brand","Pimcore\\Model\\DataObject\\Concrete","Pimcore\\Model\\DataObject","Pimcore\\Model\\DataObject\\AbstractObject","Pimcore\\Model\\Element\\AbstractElement","Pimcore\\Model\\AbstractModel"],"interfaces":["Pimcore\\Model\\DataObject\\LazyLoadedFieldsInterface","Stringable","Pimcore\\Model\\Element\\DirtyIndicatorInterface","Pimcore\\Model\\Element\\ElementDumpStateInterface","Pimcore\\Model\\Element\\ElementInterface","Pimcore\\Model\\ModelInterface"]},"allowInheritance":false,"allowVariants":false,"showVariants":false,"showAppLoggerTab":false,"showFieldLookup":false,"linkGeneratorReference":"","o_modificationDate":1631621994,"o_userModification":2,"versionDate":1631621994,"versionCount":7,"icon":null,"iconCls":"pimcore_icon_object","cls":" ","qtipCfg":{"title":"ID: 14","text":"Type: Brand"}},"layout":{"fieldtype":"panel","layout":null,"border":false,"name":"pimcore_root","type":null,"region":null,"title":null,"width":0,"height":0,"collapsible":false,"collapsed":false,"bodyStyle":null,"datatype":"layout","permissions":null,"childs":[{"fieldtype":"panel","layout":null,"border":false,"name":"Layout","type":null,"region":null,"title":"","width":"","height":"","collapsible":false,"collapsed":false,"bodyStyle":"","datatype":"layout","permissions":null,"childs":[{"fieldtype":"input","width":"","defaultValue":null,"queryColumnType":"varchar","columnType":"varchar","columnLength":190,"regex":"","unique":false,"showCharCount":false,"name":"name","title":"Brand Name","tooltip":"","mandatory":false,"noteditable":false,"index":false,"locked":false,"style":"","permissions":null,"datatype":"data","relationType":false,"invisible":false,"visibleGridView":false,"visibleSearch":false},{"fieldtype":"numeric","width":"","defaultValue":null,"queryColumnType":"double","columnType":"double","integer":false,"unsigned":false,"minValue":null,"maxValue":null,"unique":false,"decimalSize":null,"decimalPrecision":null,"name":"ranking","title":"Brand Ranking","tooltip":"","mandatory":false,"noteditable":false,"index":false,"locked":false,"style":"","permissions":null,"datatype":"data","relationType":false,"invisible":false,"visibleGridView":false,"visibleSearch":false},{"fieldtype":"image","queryColumnType":"int(11)","columnType":"int(11)","name":"logo","title":"Brand Logo","tooltip":"","mandatory":false,"noteditable":false,"index":false,"locked":false,"style":"","permissions":null,"datatype":"data","relationType":false,"invisible":false,"visibleGridView":false,"visibleSearch":false,"width":"","height":"","uploadPath":""},{"fieldtype":"objectbricks","allowedTypes":["myLittleBrick"],"maxItems":"","border":false,"name":"brandBrick","title":"Brand Brick","tooltip":"","mandatory":false,"noteditable":false,"index":false,"locked":false,"style":"","permissions":null,"datatype":"data","relationType":false,"invisible":false,"visibleGridView":false,"visibleSearch":false}],"locked":false,"icon":"","labelWidth":0,"labelAlign":"left"}],"locked":false,"icon":null,"labelWidth":100,"labelAlign":"left"},"userPermissions":{"save":true,"unpublish":true,"lEdit":true,"lView":true,"layouts":true,"list":true,"view":true,"publish":true,"delete":true,"rename":true,"create":true,"settings":true,"versions":true,"properties":true},"versions":[{"id":54,"cid":14,"ctype":"object","note":"","date":1631621994,"public":false,"versionCount":7,"autoSave":false,"user":{"name":"admin","id":2},"index":0}],"scheduledTasks":[],"childdata":{"id":14,"data":{"classes":[],"general":{"o_classId":"brand","o_className":"Brand","o_published":true,"o_id":14,"o_parentId":13,"o_type":"object","o_key":"Xbox","o_creationDate":1630566211,"o_userOwner":2,"fullpath":"\/Brands\/Xbox","o_locked":false,"php":{"classes":["Pimcore\\Model\\DataObject\\Brand","Pimcore\\Model\\DataObject\\Concrete","Pimcore\\Model\\DataObject","Pimcore\\Model\\DataObject\\AbstractObject","Pimcore\\Model\\Element\\AbstractElement","Pimcore\\Model\\AbstractModel"],"interfaces":["Pimcore\\Model\\DataObject\\LazyLoadedFieldsInterface","Stringable","Pimcore\\Model\\Element\\DirtyIndicatorInterface","Pimcore\\Model\\Element\\ElementDumpStateInterface","Pimcore\\Model\\Element\\ElementInterface","Pimcore\\Model\\ModelInterface"]},"allowInheritance":false,"allowVariants":false,"showVariants":false,"showAppLoggerTab":false,"showFieldLookup":false,"linkGeneratorReference":""}}},"data":{"name":"XBOX","ranking":1,"logo":{"type":"image","id":3,"parentId":2,"parent":null,"filename":"xbox.png","path":"\/_default_upload_bucket\/","mimetype":"image\/png","creationDate":1630566244,"modificationDate":1630566244,"stream":null,"userOwner":2,"userModification":2,"properties":null,"versions":null,"metadata":[],"locked":null,"customSettings":{"imageDimensionsCalculated":true,"imageWidth":225,"imageHeight":225,"embeddedMetaData":{"FileSize":"2.7 kB","FileModifyDate":"2021:09:02 07:04:04+00:00","FileAccessDate":"2021:09:02 07:04:04+00:00","FileInodeChangeDate":"2021:09:02 07:04:04+00:00","FilePermissions":"rw-rw-rw-","FileType":"PNG","FileTypeExtension":"png","MIMEType":"image\/png","ImageWidth":225,"ImageHeight":225,"BitDepth":8,"ColorType":"Palette","Compression":"Deflate\/Inflate","Filter":"Adaptive","Interlace":"Noninterlaced","Palette":"(Binary data 144 bytes, use -b option to extract)","ImageSize":"225x225","Megapixels":0.051},"embeddedMetaDataExtracted":true},"hasMetaData":false,"siblings":null,"hasSiblings":null,"scheduledTasks":null,"_dataChanged":false,"versionCount":1,"dependencies":null,"__dataVersionTimestamp":1630566244,"dao":null,"_fulldump":false,"o_dirtyFields":null,"____pimcore_cache_item__":"asset_3"},"brandBrick":[{"data":{"someValue":"asdf","someBlock":[{"oIndex":0,"data":{"someNumber":1,"someSelect":"three"}}]},"type":"myLittleBrick","metaData":{"someValue":{"objectid":14,"inherited":false},"someBlock":{"objectid":14,"inherited":false}},"inherited":false,"title":""}]},"metaData":{"name":{"objectid":14,"inherited":false},"ranking":{"objectid":14,"inherited":false},"logo":{"objectid":14,"inherited":false},"brandBrick":{"objectid":14,"inherited":false}},"properties":[],"validLayouts":[{"id":-1,"name":"Master (Admin Mode)"},{"id":0,"name":"Master"}],"currentLayoutId":"-1","workflowManagement":{"hasWorkflowManagement":false}}<br />
<b>Fatal error</b>:  Uncaught TypeError: Cannot assign Pimcore\Model\Element\ElementDescriptor to property Pimcore\Model\DataObject\Objectbrick::$object of type ?Pimcore\Model\DataObject\Concrete in /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php:252
Stack trace:
#0 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): ReflectionProperty-&gt;setValue(Object(Pimcore\Model\DataObject\Brand\BrandBrick), Object(Pimcore\Model\Element\ElementDescriptor))
#1 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Brand\BrandBrick), Object(ReflectionProperty))
#2 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Brand\BrandBrick))
#3 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Brand\BrandBrick))
#4 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Data\BlockElement), Object(ReflectionProperty))
#5 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Data\BlockElement))
#6 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(94): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#7 /var/www/vendor/pimcore/pimcore/models/DataObject/Data/BlockElement.php(246): DeepCopy\DeepCopy-&gt;copy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#8 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(594): Pimcore\Model\DataObject\Data\BlockElement-&gt;marshalForCache()
#9 [internal function]: Pimcore\Cache\Core\CoreCacheHandler-&gt;Pimcore\Cache\Core\{closure}(Object(Pimcore\Model\DataObject\Data\BlockElement))
#10 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/TypeFilter/ReplaceFilter.php(28): call_user_func(Object(Closure), Object(Pimcore\Model\DataObject\Data\BlockElement))
#11 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(125): DeepCopy\TypeFilter\ReplaceFilter-&gt;apply(Object(Pimcore\Model\DataObject\Data\BlockElement))
#12 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(155): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#13 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(135): DeepCopy\DeepCopy-&gt;copyArray(Array)
#14 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(155): DeepCopy\DeepCopy-&gt;recursiveCopy(Array)
#15 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(135): DeepCopy\DeepCopy-&gt;copyArray(Array)
#16 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy-&gt;recursiveCopy(Array)
#17 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Objectbrick\Data\MyLittleBrick), Object(ReflectionProperty))
#18 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Objectbrick\Data\MyLittleBrick))
#19 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Objectbrick\Data\MyLittleBrick))
#20 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Brand\BrandBrick), Object(ReflectionProperty))
#21 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Brand\BrandBrick))
#22 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Brand\BrandBrick))
#23 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy-&gt;copyObjectProperty(Object(Pimcore\Model\DataObject\Brand), Object(ReflectionProperty))
#24 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy-&gt;copyObject(Object(Pimcore\Model\DataObject\Brand))
#25 /var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(94): DeepCopy\DeepCopy-&gt;recursiveCopy(Object(Pimcore\Model\DataObject\Brand))
#26 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(605): DeepCopy\DeepCopy-&gt;copy(Object(Pimcore\Model\DataObject\Brand))
#27 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(910): Pimcore\Cache\Core\CoreCacheHandler-&gt;storeCacheData('object_14', Object(Pimcore\Model\DataObject\Brand), Array, NULL, false)
#28 /var/www/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(958): Pimcore\Cache\Core\CoreCacheHandler-&gt;writeSaveQueue()
#29 /var/www/vendor/pimcore/pimcore/lib/Cache.php(200): Pimcore\Cache\Core\CoreCacheHandler-&gt;shutdown(false)
#30 /var/www/vendor/pimcore/pimcore/lib/Pimcore.php(242): Pimcore\Cache::shutdown()
#31 /var/www/vendor/pimcore/pimcore/lib/Kernel.php(247): Pimcore::shutdown()
#32 [internal function]: Pimcore\Kernel-&gt;Pimcore\{closure}()
#33 {main}
  thrown in <b>/var/www/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php</b> on line <b>252</b><br />
```

![image](https://user-images.githubusercontent.com/20439617/133258845-6f273806-3bfa-4478-8689-41ad8a84ca44.png)

Placing Blocks into **Field Collections** does not result in any errors.

**Comment by brusch** [comment]
Fixed by #10319
